### DISCUSSION TITLE
渺软公益 CDN

### DISCUSSION THREAD

**Discussion by zkeq** [zkeq/Coding#644] [Announcements]
渺软公益 CDN 正式对外开放啦
初版页面只适配了电脑端（1000px+ 屏幕）
https://cdn.onmicrosoft.cn/

**Comment by zkeq** [comment]
#### Services Terms of Use

本项目的开发初衷是为了方便开发者使用静态的 `js` `css` 等文件，**不支持**托管私人静态图片 私人字体文件等大流量资源（公共字体资源会转发至子项目 `font.on` ）

同时这也是违反 `jsdelivr` 的 `TOS` 的:https://www.jsdelivr.com/terms/acceptable-use-policy-jsdelivr-net

```markdown
2. Abusing the service and its resources, or using jsDelivr as a general-purpose file or media hosting service. This includes, for example:

- running an image hosting website and using jsDelivr as a storage for all uploaded images,
- hosting videos, file backups, or other files in large quantities.

We recognize that there are legitimate projects that consist of a large number of files, and these are not considered abuse. For example: icons packs, apps, or games with a large number of assets.
```

以及同样违反 `npmjs.com` 的 `TOS`: https://docs.npmjs.com/policies/open-source-terms#acceptable-content

```markdown
4. Packages that are not functionally compatible with the npm command-line client. 
For example, a "package" cannot simply be a PNG or JPEG image, a movie file, or a text document uploaded directly to the registry. 
Using the Public Registry as a general purpose database is not allowed.
```

若您有托管图片的需求，推荐使用由 [@杜老师](https://dusays.com/) 运营的 `去不图床`，同样使用国内CDN `多吉云` 进行加速

- https://7bu.top

若您有请求大字体文件的需求，请考虑 [Google Fonts](https://fonts.google.com/) ，以及对应的国内镜像

- https://cdn.baomitu.com/index/fonts

若您对我们的处理结果有异议，请联系邮箱📮： cdn@onmicrosoft.cn

上述所有域名 若由于不可抗力出现中断、暂停等情况，渺软不对此域名的 SLA 负责。

> 更新于：2023年1月16日 17:21

**Reply by linlinzzo** [reply]
杜老师的图床要付费，如果想要快速、免费、稳定的图床的话，建议使用 https://img.pho.pub 。这个图床需要联系管理员审核，空间不够的话在一定范围内可以扩容。

**Reply by zkeq** [reply]
判定规则是这样的：公共库类型的，都可以加速，但是自己私库的不支持。

就比如字体，主题的 logo 之类的，是开放给公共使用的开源静态资源，如此可以加速。

但是如果你是自己的私库，别人用到的 概率为 0 ，很抱歉 本项目不支持此类加速。

**Comment by zkeq** [comment]
这里汇总一下可用的国内的静态CDN加速公益服务

大厂运营：

- 饿了么（未公开发布）：https://npm.elemecdn.com/
- 七牛云存储: ~~https://staticfile.org/~~ (谨慎选择)
- 360 前端静态资源库: https://cdn.baomitu.com/
- 字节跳动静态资源公共库: https://cdn.bytedance.com/
- 又拍云: http://jscdn.upai.com
- 新浪云计算CDN公共库:：https://lib.sinaapp.com/
- 极兔云 联合 Bootstrap 中文网 ：~~https://www.bootcdn.cn/~~ (谨慎选择)
- cdnjs 相关：https://github.com/cdnjs/cdnjs/discussions/14115

个人运营：

- ~~https://tianli-blog.club/jsd/~~
- https://u.sb/css-cdn/
- https://www.sourcegcdn.com
- https://bilicdn.tk
- https://blog.1id.top/post/zi-zhi-de-yi-ge-jsdelivr-jing-xiang-fen-xiang
- https://www.xxhzm.cn/archives/733/
- https://cdn.haorwen.tk
- https://jsew.cky.codes
- https://a-js.cc/archives/128.html
- https://github.com/54ayao/Chinajsdelivr
- https://jsd.cdn.zzko.cn/
- https://cdn.akass.cn
- https://cdn2.chuqis.com

知名院校以及非盈利机构运营

- 南方科技大学：https://mirrors.sustech.edu.cn/help/cdnjs.html

* 注：标记 `谨慎选择` 的服务商，存在过恶意投毒事件，如需使用请加 `SRI`。

> 更新于 2024年07月11日 21:19 （在 [此条评论](https://github.com/zkeq/Coding/discussions/644#discussioncomment-4571515) 补充可跟进）

**Reply by codejiangqihan** [reply]
【 已收录 】补充一个：https://www.codeqihan.com/post/zi-zhi-de-yi-ge-jsdelivr-jing-xiang-fen-xiang/

**Reply by lbnliboning** [reply]
补充一个https://cdn.1703.site（前几天因兼容原因暂停服务了，现在好了）

**Reply by Ryan83602** [reply]
支持,本人为[Moezz CDN](https://cdn.moezz.cn)维护者(e,基本上就是一个人干几个人的活),支持这种CDN!

**Reply by ocoke** [reply]
`jsew.cky.codes` 新增中国大陆节点，请合理使用：`s.cky.qystu.cc`

**Reply by akatsukiro** [reply]
您好，关于 `img.chs.pub (cdn.akass.cn)` 相关事宜我已于3月8日通过邮件进行了回复，长时间没有等到您的答复，我便只好来这里留个言，期待得到您的回复。

**Reply by zkeq** [reply]
> 您好，关于 `img.chs.pub` 相关事宜我已于3月8日通过邮件进行了回复，长时间没有等到您的答复，我便只好来这里留个言，期待得到您的回复。

收到了 最近在忙其他的 没看邮件抱歉

**Reply by Fgaoxing** [reply]
[收到] CDNN维护: https://cdn2.chuqis.com

**Reply by Fgaoxing** [reply]
[已跟进] 帮Tianli留个言，TianliCDN停止维护，后续流量转到cdn.chuqis.com

**Reply by lainbo** [reply]
补充一个
https://www.webcache.cn/

**Reply by xingpingcn** [reply]
字体加载不了https://font.onmicrosoft.cn/@fortawesome/fontawesome-free@6.1/webfonts/fa-solid-900.woff2

**Reply by Young-Lord** [reply]
建议立刻移除 https://www.bootcdn.cn/ 与 https://staticfile.org/ 并作出警告！这两个镜像源已被发现有篡改js文件，插入恶意代码行为。
https://sansec.io/research/polyfill-supply-chain-attack

**Reply by zkeq** [reply]
> 建议立刻移除 [bootcdn.cn](https://www.bootcdn.cn/) 与 [staticfile.org](https://staticfile.org/) 并作出警告！这两个镜像源已被发现有篡改js文件，插入恶意代码行为。 [sansec.io/research/polyfill-supply-chain-attack](https://sansec.io/research/polyfill-supply-chain-attack)

你的引用来源是 文章里面讲的是 `Polyfill.js` 这个库的一些问题 和这两个镜像站有什么关系?

**Reply by Young-Lord** [reply]
> 你的引用来源是 文章里面讲的是 `Polyfill.js` 这个库的一些问题 和这两个镜像站有什么关系?

首先，说的并不是`Polyfill.js`库本身有问题，而是使用`polyfill.io`等域名的CDN向js文件中投毒。

文章最上面的`Update June 28th`已经指明这几个站点均有恶意行为，并且另外几篇文章也具体指出这些站点是由同一个团体操控的： https://www.bleepingcomputer.com/news/security/polyfillio-bootcdn-bootcss-staticfile-attack-traced-to-1-operator/
而且更早时候，也有人发现了这个CDN的投毒现象： https://www.v2ex.com/t/950163

**Reply by TechQuery** [reply]
@zkeq @Young-Lord 不是《金融时报》polyfill.io 原团队下的毒，而是他们不维护之后域名卖给中国 CDN 服务商之后出现的问题：formatjs/formatjs#4363

**Reply by zkeq** [reply]
已在显著位置标明

**Comment by zkeq** [comment]
#### Update Logs

- ~~`2024年04月21日` ：【业务】得到 图灵SCDN https://scdn.qqvip.com 赞助, 项目国内域名全部切至该服务 .~~

- `2024年04月02日` ：【业务】迫于学业与生活压力, 该项目所有域名停止运行维护.

- `2023年09月18日` ：【业务】特殊业务需求, `jsd.on`、`npm.on` 404状态码缓存时间刷新时间由原来的 `10s` 更改至 `180天`

- `2023年08月20日` ：【项目】自建源站于上海腾讯云服务器, 临时避险时启用.

- `2023年08月20日` ：【业务】屡次被攻击, 国内CDN以及源站, 紧急避险下线所有域名.
	- 相关附件: https://bu.dusays.com/2023/08/20/64e11df890195.png
	- 相关附件: https://bu.dusays.com/2023/08/20/64e11dfc988a3.png
	- 相关附件: https://bu.dusays.com/2023/08/20/64e11df92de50.png
	- 相关附件: https://bu.dusays.com/2023/08/20/64e11dfaedd79.png
	- 相关附件: https://bu.dusays.com/2023/08/20/64e11df9d9dc1.png
	- 相关附件: https://bu.dusays.com/2023/08/20/64e11f6205add.png

- `2023年08月17日` ：【业务】与 [初七云](https://www.chuqiyun.com/) 达成合作, `*.cmds.run` 域名业务全部上线恢复 ~ 🎆

- `2023年08月16日` ：【项目】项目被 [夏柔(免费API)](https://api.aa1.cn/) `名人堂` 收录 ~, 被冠名以 `知名作者` 称号 🚀, 相关链接: [作者首页](https://api.aa1.cn/author/20) [热榜地址](https://api.aa1.cn/hot)
	- 相关附件: https://bu.dusays.com/2023/08/16/64dcb9c5cca48.png 
	- 相关附件: https://bu.dusays.com/2023/08/16/64dcbb05564f5.png

- `2023年08月16日` ：【业务】因部分资源拉取源站响应缓慢, 将3个域名的缓存有效期从2个月提升至180天(半年), 保证服务的可用性 ~

- `2023年08月13日` ：【业务】`[火山引擎]` font.on 域名遭受上线以来最大流量攻击, 损失流量 `6.68` TB, 详细信息: 首先收到欠费短信(1), 垫交所描述欠费余额后上线海外分流业务, 但是随之而来的是欠费更多的信息(2), 随之而来的是看到同一时间 Cloudflare 被攻击的信息(3), 采取封禁 Url 的策略(4)(5),暂时全线切至 `CloudFlare`, 后续业务待定(6). 后续: 攻击持续至13日晚7点, 子业务(追星计划)被盗刷流量 600GB (9) - 多吉云扣费账单(11), `CloudFlare`等子业务再次被刷流量 9TB (10). 
	- 相关附件: https://bu.dusays.com/2023/08/13/64d81ffbb19ba.png
	- 相关附件(1): https://bu.dusays.com/2023/08/13/64d8205aace8c.jpeg
	- 相关附件(2): https://bu.dusays.com/2023/08/13/64d820c43a729.png
	- 相关附件(3): https://bu.dusays.com/2023/08/13/64d8213de323f.png
	- 相关附件(4): https://bu.dusays.com/2023/08/13/64d8219602952.jpg
	- 相关附件(5): https://bu.dusays.com/2023/08/13/64d821aee4ef5.jpg
	- 相关附件(6): https://bu.dusays.com/2023/08/13/64d821fc8bf9e.png
	- 相关附件(7): https://bu.dusays.com/2023/08/13/64d831606dccc.png
	- 相关附件(8): https://bu.dusays.com/2023/08/14/64d956cdf1420.png
	- 相关日志: https://pan.baidu.com/s/1dT1LIPGwHp4jCDQcb4Yg1g?pwd=zkeq
 	- 相关附件(9): https://bu.dusays.com/2023/08/14/64d974a17a10d.png
	- 相关附件(10): https://bu.dusays.com/2023/08/14/64d9750db6c05.png
	- 相关附件(11): https://bu.dusays.com/2023/08/14/64d98c1583eb6.png
	- 相关附件(12): https://bu.dusays.com/2023/08/18/64def2a66f3ce.png

- `2023年07月20日`  ：【业务】`*.cmds.run` 域名由于赞助商 `网盾星球` 业务调整, 取消赞助计划, 该域名已全线切回 `CloudFlare` 因 用量/需求量 过少, 故该子业务暂时搁置~

- `2023年07月20日`  ：【业务】**官网API展示业务中断10h** 未知原因 `官网API` 服务**短暂中断**, 已切换备用回源恢复业务的可用性 (**[实际业务](https://status.onmicrosoft.cn/)未受影响**)
	- 相关附件： https://bu.dusays.com/2023/07/20/64b93272d5529.png
	- 相关附件： https://bu.dusays.com/2023/07/20/64b931ea40ae8.jpg
	- 相关附件： https://bu.dusays.com/2023/07/20/64b931ea40bda.jpg

- `2023年07月02日`  ：【项目】与 `shssedu.ac.cn` 作者取得联系, 将 `/img` 目录转发至独立子项目
 	- 相关附件： https://bu.dusays.com/2023/07/02/64a18a8921b3b.png

- `2023年07月01日`  ：【项目】`shssedu.ac.cn` 引用本项目, 但由于该网站图片均为未压缩的大图片, 导致CDN流量激增, 业务上线图片压缩函数, 效果显著 `1.2M --> 77KB`
	- 相关附件： https://bu.dusays.com/2023/07/01/649fe02b4fd87.png
	- 相关附件： https://bu.dusays.com/2023/07/01/649fe0199ba04.png
	- 相关附件： https://bu.dusays.com/2023/07/01/649fe03985a78.png
	- 相关附件： https://bu.dusays.com/2023/07/01/649fe07962aae.png
	- ~~相关附件： https://bu.dusays.com/2023/07/01/649ff103cf372.png~~
	- 相关附件： https://bu.dusays.com/2023/08/13/64d8794511da6.png
	- 相关附件： https://bu.dusays.com/2023/08/13/64d879c0218bf.png

- `2023年06月25日`  ：【项目】Sponsor 组件增加 `日期` 列, 该部分计划独立成子业务项目, 为广大开发者提供爱发电公共展示页面服务(待实施).

- `2023年05月10日`  ：【业务】cdnjs.on 域名首次被大流量网站使用 业务规模似乎提升 但目标源为小程序 `UA` 无法判断来源 :) 业务图表max值调高.
   - 相关附件: https://bu.dusays.com/2023/05/10/645a95ea38092.png

- `2023年05月10日`  ：【业务】业务第一次遭到灰黑产业对接, 几小时突破服务限制, 采取全面封禁策略.
   - 相关附件: https://pan.baidu.com/s/17KMZZm4yaM9C-Ypxg8pOow?pwd=zkeq

- `2023年04月27日`  ：【业务】备案主体更新至企业备案, 增加业务的可用性 备案主体为: `郑州市郑东新区渺软科技工作室`.

- `2023年03月07日`  ：【业务】自费服务器，子项目 `*.cmds.run` 业务 全部恢复正常运行

- `2023年03月03日`  ：【业务】因美国服务器到期 子项目： `*.cmds.run` 域名业务全部中断，所幸业务代码得以保留，在寻找解决方案。

- `2023年02月19日`  ：【业务】分析前三次的攻击，找出疑似攻击源 IP 799 个，封禁 150 个 重点 IP
	- 相关附件： 疑似攻击源：799个: [MiaoRun Block IPs 2023-3-19.log](https://gist.cmds.run/zkeq/afe0773bf0c81a38562ee040a2ad04cb/raw/be538f67a0d3d94555b78eaebdc5a38ad5e2481f/MiaoRun%20Block%20IPs%202023-3-19.log)
	- 相关附件：三次攻击期间所有日志: [MiaoRun_Free_CDN-2023_3_19.zip](https://down.onmicrosoft.cn/vika/2023/02/19/902b1d2b352648ebbdd2e01e07302ce4?attname=MiaoRun_Free_CDN-2023_3_19.zip)
	- 相关附件：本次封禁IP：150个（在攻击时段 请求重点 Url 300 次 +）: [2023-03-19 Banned.log](https://gist.cmds.run/zkeq/afe0773bf0c81a38562ee040a2ad04cb/raw/c0680f631e4a7ded956307301330bee97117d9fe/01_2023-03-19%20Banned.log)
	- 上述证据已报警移交警方处理：https://bu.dusays.com/2023/02/19/63f2350021aae.jpg

- `2023年02月19日`  ：【业务】jsd.on 域名 再次遭受大流量攻击，本次攻击损失流量 `170GB` ，采取对其盗刷链接限速的措施，保证了业务的可用性。
	- 相关附件： https://bu.dusays.com/2023/02/19/63f1e91c25875.png
	- 相关附件： https://bu.dusays.com/2023/02/19/63f1e91c346e2.png
	- 相关附件： https://bu.dusays.com/2023/02/19/63f1e91c3d246.png
	- 相关附件： https://bu.dusays.com/2023/02/19/63f1e91c418a5.png

- `2023年02月18日`  ：【业务】cdn.on 遭到盗刷流量，损失流量 `30GB` .
	- 相关附件： https://bu.dusays.com/2023/02/19/63f1ea15afc32.png

- `2023年02月17日`  ：【业务】cdn.on jsd.on 域名分别遭到大流量攻击，损失流量 `70GB`，超过承诺服务限制，调高阈值，尽量保证业务域名的可用持久性。
	- 相关附件： https://bu.dusays.com/2023/02/17/63ef9f3d862e1.png
	- 相关附件： https://bu.dusays.com/2023/02/18/63f0248723bb3.png
	- 相关附件： https://bu.dusays.com/2023/02/19/63f1e9f6e20ea.png

- `2023年02月11日`  ：【业务】因服务屡次遭到境外大流量攻击，每次造成多吉云境外流量 30GB 左右不等
	- 此部分不包含在境内流量包内，账单直接从余额扣除，导致成本激增，现将境外 DNS 解析至 CloudFlare ，减少运营成本。

- `2023年02月10日`  ：【业务】npm.on 开启 302 跟随回源模式，但是仍然不建议在生产环境中不指定版本号。

- `2023年02月10日`  ：【项目】上线 刷新缓存(Pure Cache) 功能，后端对接 字节、多吉、网盾星球 🌍 三家 服务商，可以在本页面进行缓存的刷新～

- `2023年02月06日`  ：【业务】✨ 升级服务，上线 `cdn.cmds.run` 域名 用户加速 GitHub (Gist), Bitbucket, GitLab RAW 并提供正确的 Content-Type ，并上线监控页面。

- `2023年02月06日`  ：【业务】上线 `raw.cmds.run` 域名，用于加速 `raw.githubusercontent.com` 所属资源由 `网盾星球` ([netdun.net](https://netdun.net)) 提供（单日 500 GB 资源包）

- `2023年02月02日`  ：【业务】新增 gravatar 加速服务，归并于 jsD 域名子目录：[jsd.onmicrosoft.cn/avatar/](https://jsd.onmicrosoft.cn/avatar/e8e49ff904f51fa196a33959bf82b418)

- `2023年01月30日`  ：【项目】为后端增加逻辑：若图表数据某一条大于平均值 10 倍，那么即记录此数据为异常无效数据，修改值为 99.99 防止小鬼恶意刷流量 导致图表 📈 显示异常！

- `2023年01月30日`  ：【业务】收到邮件，过去 7 天在线率 100% 🎉
	- 相关附件： https://bu.dusays.com/2023/01/30/63d72d143a0d4.png

- `2023年01月30日`  ：【业务】增加新功能，当 jsD 请求 .css 后缀结尾的文件时，自动进行如下规则替换（防止 CSS 中写死）

```py
	.replace("fastly.jsdelivr.net", "jsd.onmicrosoft.cn")\
	.replace("cdn.jsdelivr.net", "jsd.onmicrosoft.cn")\
	.replace("gcore.jsdelivr.net", "jsd.onmicrosoft.cn")\
		\
	.replace("unpkg.com", "npm.onmicrosoft.cn")\
		\
	.replace("cdnjs.cloudflare.com", "cdnjs.onmicrosoft.cn")
```

- `2023年01月25日`  ：【业务】与 `网盾星球` ([netdun.net](https://netdun.net)) 达成合作，感谢其提供 2T 储备流量包，相关业务待部署，初步计划使用 `cmds.run` 域名 上线 `raw.githubusercontent` 公益回源业务，后者已在备案流程中。

- `2023年01月23日`  ：【业务】感谢 `Ray` 捐助域名 : `cmds.run` ，待部署新业务于此域名，以及做主域名的备用。

- `2023年01月21日`  ：【业务】与 `无畏云` (www.sctes.com) 达成合作，但与本站原有架构 `API` 等 不吻合，进一步 对接事宜 待商榷。

- `2023年01月17日`  ：【业务】因尝试自建 unpkg 回源，造成的体验极差，故继续使用 vercel 回源，本次切换节点至香港，并设立相关监控，回源可用性恢复。
	- （也感谢 夏柔公益 为本项目赞助一台 香港 CN2 回源服务器）

- `2023年01月17日`  ：【业务】第一封 Abuse 封禁通知邮件发出，邮件格式规范相关待定，此邮箱由飞书企业邮托管，可支持后续团队共享处理权限。
	- 相关附件： https://bu.dusays.com/2023/01/17/63c657697ea2d.png

- `2023年01月16日`  ：【业务】上线业务邮箱：cdn@onmicrosoft.cn，负责封禁的通知和申诉功能。

- `2023年01月14日`  ：【业务】由于使用 vercel 回源 unpkg.com 时被 CloudFlare 拦截，故自建 unpkg 源站于 icodel 海外 8H16G 大带宽主机上，用户体验无影响。
	- 相关附件： https://bu.dusays.com/2023/01/15/63c3c484033ea.jpg
	- 相关附件： https://bu.dusays.com/2023/01/15/63c3c48410fc7.png

- `2023年01月13日`  ：【业务】上线 font.onmicrosoft.cn 域名主页：https://font.onmicrosoft.cn/@info

- `2023年01月13日`  ：【业务】成功与 [夏柔公益](https://www.aa1.cn/) 达成合作，稳定上线 font.onmicrosoft.cn 域名。

- `2023年01月13日`  ：【项目】上线 font.on 大流量 Abuse 面板，后端对接 火山引擎 审核后自动完成 封禁 解封 等功能。

- `2023年01月13日`  ：【业务】完成 font.onmicrosoft.cn 域名部署，该域名仅负责加速字体文件，其他重定向至官网，请勿使用此域名加速非 .css .woff2 .woff .ttf .otf 后缀的文件
	- 相关附件： https://bu.dusays.com/2023/01/13/63c0bd3f2b79f.png

- `2023年01月12日`  ：【业务】上线 Status 页面，内置事件发布功能

- `2023年01月12日`  ：【业务】.woff2 .woff .ttf .otf 等后缀 ~~转发至 npm.elemecdn.com 计划待该服务不可用时会上线提供专用字体加速域名~~ 转发至 font.onmicrosoft.cn，新增 转发 tag。
	- 相关附件： https://bu.dusays.com/2023/01/13/63c0bc6a5a1da.png
	- 相关附件： https://bu.dusays.com/2023/01/13/63c0bcdddfb31.png

- `2023年01月11日`  ：【业务】项目业务调整 官网下线 unpkg.on 域名，新增 cdnjs.on 域名，提供 cdnjs 的加速服务，unpkg.on 域名采用 302 的方式过渡到 npm.on 域名
	- 相关附件： https://bu.dusays.com/2023/01/13/63c0bcbd21fcb.png

- `2023年01月10日`  ：【项目】[xcsoft](https://github.com/soxft) 基于 React + TS 对本项目的 重构 / 重写 完成，项目成功完成上线工作(创世更新)！

- `2023年01月08日`  ：【业务】杜老师说 发文推广本站，更多人开始使用！感谢杜老师！

- `2023年01月08日`  ：【业务】更新服务条款，改为 **不支持**托管加速 私人字体、私人图片

- `2023年01月06日`  ：【项目】Abuse 系统开发完毕！上线于项目原型，对接 X OpenID 实现管理员登陆功能
	- 相关附件： https://bu.dusays.com/2023/01/11/63bece3880099.png
	- 相关附件： https://bu.dusays.com/2023/01/11/63becee27718b.png

- `2023年01月05日`  ：【项目】项目前端 重构 / 重写 初始提交！

- `2023年01月04日`  ：【项目】开始开发在线 Abuse 系统

- `2023年01月02日`  ：【业务】书写本站 Services Terms of Use ，明确表示不支持 字体、图片 的加速服务。

- `2023年01月02日`  ：【业务】后台日志发现有站已经滥用本服务长达数月，流量 Top 30 有近一半都是私人图片，决定开发 Abuse 系统。

- `2023年01月01日`  ：【项目】成功注册萌备号：20231314，与本站备案号 2022013147 相呼应！

- `2022年12月31日`  ：【项目】项目原型成功上线！提供 jsd.on npm.on unpkg.on 三个域名加速服务。

- `2022年12月27日`  ：【项目】前端设计图画出，[zkeq](https://github.com/zkeq) 尝试用 vue3 + antdv + element 开发一套前端项目原型。
	- 相关附件： https://bu.dusays.com/2023/01/11/63becb5d4a3b7.png

- `2022年09月04日`  ：【项目】后端基础核心功能开发完毕（图表相关数据以及核心缓存机制）

- `2022年08月13日`  ：【业务】意识到如果想要做公益服务，那么就要有一套对外公开的数据相关，项目总思路确立，第一次后端提交于`zkeq\DogeCloud`

- `2022年07月18日`  ：【业务】攻击最终以封禁对应 url 并且采取一些反制措施结尾 本次攻击共接受请求数: 31120893, 损失流量: 32.14GB
	- 相关附件： https://bu.dusays.com/2023/01/11/63becb22bd0e6.png
	- 相关附件： https://bu.dusays.com/2023/01/11/63becb22bd173.png

- `2022年07月15日`  ：【业务】受到不明流量攻击，经查本站属于误伤，攻击源也明确。

- `2022年05月31日`  ：【项目】jsd.onmicrosoft.cn 正式开始解析，并小范围投入使用。


> 更新于 2024年04月02日 12:17

**Reply by TechQuery** [reply]
活菩萨，我刚要把所有项目的 JSDelivr、UnPkg 换成 onmicrosoft.cn，您就停服了😭

有没有开源代码？我去薅资源部署个新的，不知道我为数不多的 CDN 镜像站经验够不够：
1. JSCIG/es-discuss#34
2. https://kaiyuanshe.feishu.cn/wiki/A1JSwFP0ti44QTkhGqncTQMYnDb

**Reply by zkeq** [reply]
> 活菩萨，我刚要把所有项目的 JSDelivr、UnPkg 换成 onmicrosoft.cn，您就停服了😭
> 
> 有没有开源代码？我去薅资源部署个新的，不知道我为数不多的 CDN 镜像站经验够不够：
> 
> 1. [JSCIG 第八次会议 JSCIG/es-discuss#34](https://github.com/JSCIG/es-discuss/issues/34)
> 2. [kaiyuanshe.feishu.cn/wiki/A1JSwFP0ti44QTkhGqncTQMYnDb](https://kaiyuanshe.feishu.cn/wiki/A1JSwFP0ti44QTkhGqncTQMYnDb)

这边对接的是多吉云的 API和统计数据全是... 换其他服务商相当于项目重写了

**Reply by TechQuery** [reply]
> 这边对接的是多吉云的 API和统计数据全是... 换其他服务商相当于项目重写了

具体的厂商 API 都是副作用代码，只要架构是清晰的，做通用化重构应该还是好移植的。

**Reply by zkeq** [reply]
> > 这边对接的是多吉云的 API和统计数据全是... 换其他服务商相当于项目重写了
> 
> 具体的厂商 API 都是副作用代码，只要架构是清晰的，做通用化重构应该还是好移植的。

你说的代码是哪方面的代码？源站转发还是后端数据处理/封禁还是前端页面

**Reply by TechQuery** [reply]
> 你说的代码是哪方面的代码？源站转发还是后端数据处理/封禁还是前端页面

除了“封禁”我没想到，其它我都期望开源。

**Reply by zkeq** [reply]
> 除了“封禁”我没想到，其它我都期望开源。

完整项目目前暂无开源计划

**Comment by zkeq** [comment]
.keep

**Reply by zkeq** [reply]
👀 感谢支持！

本页面评论区仅做通知用，其他内容发到博客哈： https://icodeq.com/2022/5c18919d3591/

**Comment by Francklin** [comment]
没有了渺软公益CDN，还有web缓存网www.webcache.cn，继续给公益蓄力。

**Reply by zkeq** [reply]
你这个支持的库好少

**Reply by TechQuery** [reply]
> 你这个支持的库好少

对呀，感觉是个白名单。@zkeq 作者大大能不能把你的代码开源，让其他人可以随意部署？

**Reply by zkeq** [reply]
> > 你这个支持的库好少
> 
> 对呀，感觉是个白名单。@zkeq 作者大大能不能把你的代码开源，让其他人可以随意部署？

即使开源 相关封禁功能用的也是多吉云 以及图表 统计数据 都是和多吉云挂钩的[图1] 想用其他平台不太可能 

设立本项目初衷就是因为当时 `https` 请求数是免费的 所以加速这些小文件只出一个流量费就可以了 成本很小

1TB 价格也就100块钱, 但是现在请求数收费 光请求数 `1TB` 就要 200 块钱 所以项目暂时处于一个停摆状态

你这边是怎么考虑的 也用多吉云?
 
![image](https://github.com/zkeq/Coding/assets/62864752/bfac7d7a-9dac-4ca0-ba0d-7e4f0aac6520)

**Reply by TechQuery** [reply]
> 即使开源 相关封禁功能用的也是多吉云 以及图表 统计数据 都是和多吉云挂钩的[图1] 想用其他平台不太可能

我不知道这个云厂商，是你直接引用了他们的 SDK 而不好移植吗？

我们有一点 CDN 的托管经验：
1. wessberg/Polyfiller#71
2. https://kaiyuanshe.feishu.cn/wiki/A1JSwFP0ti44QTkhGqncTQMYnDb

**Reply by zkeq** [reply]
> > 即使开源 相关封禁功能用的也是多吉云 以及图表 统计数据 都是和多吉云挂钩的[图1] 想用其他平台不太可能
> 
> 我不知道这个云厂商，是你直接引用了他们的 SDK 而不好移植吗？
> 
> 我们有一点 CDN 的托管经验：
> 
> 1. [feat: add Docker configuration with Object Storage support wessberg/Polyfiller#71](https://github.com/wessberg/Polyfiller/pull/71)
> 2. https://kaiyuanshe.feishu.cn/wiki/A1JSwFP0ti44QTkhGqncTQMYnDb

功能问题，据我所知支持URL/refer封禁的厂商只有多吉 本项目是在cdn侧进行的封禁

我的意思是本站的功能性代码全是和多吉云绑定的 如果你这边不用多吉云而是自己对接 那工作量就相当于重写 开源没有必要

**Reply by Francklin** [reply]
主要是github等代码托管下的代码太乱，设置了白名单，可申请收录，也为了项目能获得久一些。

**Reply by TechQuery** [reply]
> 主要是github等代码托管下的代码太乱

@Francklin 我们说的不是 GitHub 的代码，而是 NPM 的前端库，它上面可没有什么乱七八糟的，有官方也会及时删除，和代码库不一样。

**Reply by Francklin** [reply]
npm库我是全部支持的呀，只设置了黑名单，不会是支持的少啊。

**Reply by zkeq** [reply]
> npm库我是全部支持的呀，只设置了黑名单，不会是支持的少啊。

https://npm.webcache.cn/vue@3.1.2/dist/vue.global.js

403 你这边是限制 refer 了?

**Reply by Francklin** [reply]
不允许直接访问，需要refer

**Reply by Francklin** [reply]
可在测试页面输入资源地址访问 https://www.webcache.cn/test.html

**Reply by Francklin** [reply]
也可在地址转换页的目标地址 直接粘贴后访问

**Reply by zkeq** [reply]
> 不允许直接访问，需要refer

本项目有些请求是不带 `refer` 的 如果您这边开放限制 可以转发到你这边

流量的话 `npm` 还好 `1/10` 吧 主要在 `jsd`

**Reply by Francklin** [reply]
承受不起啊，我需要根据refer内容来限制流量，无refer将会被盗刷大量流量。你看能不能想办法加上refer

**Reply by Francklin** [reply]
我们不会反代JSD，而是要逐步实现JSD的主要功能。

**Reply by zkeq** [reply]
> 我们不会反代JSD，而是要逐步实现JSD的主要功能。

这边后台看了下, 切入 `cf` 之后因为没有访问限制 流量已经达到 `2T/月` 了...

**Reply by Francklin** [reply]
达到免费的极限了吗？

**Reply by zkeq** [reply]
> 达到免费的极限了吗？

还早 只是要切回来工作量有点大

**Reply by Francklin** [reply]
> > 达到免费的极限了吗？
> 
> 还早 只是要切回来工作量有点大

我看贵站不是全部切到图灵SCDN了吗？是最近有想法了？

**Reply by zkeq** [reply]
> 我看贵站不是全部切到图灵SCDN了吗？是最近有想法了？

他们给的是一个月 1TB 的流量套餐 用不了几天就要换回来 来回切换 有点麻烦 而且那边没有能自定义封禁的功能 做不了限制

**Reply by Francklin** [reply]
> > 我看贵站不是全部切到图灵SCDN了吗？是最近有想法了？
> 
> 他们给的是一个月 1TB 的流量套餐 用不了几天就要换回来 来回切换 有点麻烦 而且那边没有能自定义封禁的功能 做不了限制

还是找些靠谱的大厂商吧，然后做严格的限制。

**Reply by Francklin** [reply]
> > 不允许直接访问，需要refer
> 
> 本项目有些请求是不带 `refer` 的 如果您这边开放限制 可以转发到你这边
> 
> 流量的话 `npm` 还好 `1/10` 吧 主要在 `jsd`

打扰了，最近在研究这个无refer方案，你能否简单介绍下你的无refer场景，为什么不能带refer，谢谢！

**Comment by michaelliunsky** [comment]
**The SSL certificate for jsd.onmicrosoft.cn has expired**.
Hope it can be fixed.Thanks!

**Reply by zkeq** [reply]
> **The SSL certificate for jsd.onmicrosoft.cn has expired**. Hope it can be fixed.Thanks!

Switch cdn forgot to update.. new certificate has been deployed

**Comment by youyou-sudo** [comment]
我看了下网站挂的爱发电域名是 `afdian.net` ，这个域名已经无法访问了；可以更换为 `afdian.com`

**Reply by Francklin** [reply]
我重新查询了下，爱发电在国内被下架了，afdian.com并没有取得国内备案号。

**Reply by TechQuery** [reply]
> 我重新查询了下，爱发电在国内被下架了，afdian.com并没有取得国内备案号。

That's sad...（难道有人看不惯用爱发电？

**Reply by zkeq** [reply]
还有这事?

**Reply by Francklin** [reply]
大概发现爱发电存在内容监管失责和运营不太合规等问题，比如缴税问题。

**Comment by TechQuery** [comment]
@zkeq 最近一周 https://npm.onmicrosoft.cn/ 又挂了

**Reply by zkeq** [reply]
> @zkeq 最近一周 [npm.onmicrosoft.cn](https://npm.onmicrosoft.cn/) 又挂了

近期是 UNPKG 官网炸了 无法进行正常回源的问题

**Reply by TechQuery** [reply]
> 近期是 UNPKG 官网炸了 无法进行正常回源的问题

他们封禁了代理？

**Reply by zkeq** [reply]
> 他们封禁了代理？

UNPKG 官方现在就是一个不可用的状态 (我在休假

**Reply by TechQuery** [reply]
> UNPKG 官方现在就是一个不可用的状态

为什么我前两天切回去就一切正常呢？

**Reply by zkeq** [reply]
> 为什么我前两天切回去就一切正常呢？

现在应该恢复了

**Reply by TechQuery** [reply]
> 现在应该恢复了

`npm.onmicrosoft.cn` 还是报 `ERR_CONNECTION_CLOSED`。

**Reply by zkeq** [reply]
> `npm.onmicrosoft.cn` 还是报 `ERR_CONNECTION_CLOSED`。

我来排查下

**Reply by zkeq** [reply]
> `npm.onmicrosoft.cn` 还是报 `ERR_CONNECTION_CLOSED`。

已恢复
